cmd/controller.go

```diff
@@ -101,15 +117,37 @@ func newControllerCommand() *cobra.Command {
 			ctrl := controller.NewController(k8sClient, dynamicClient, argocdService, namespace, appLabelSelector, registry)
 			err = ctrl.Init(context.Background())
 			if err != nil {
-				return err
+				panic(err)
+			}
+
+			if leaderElectionConfiguration.LeaderElect {
+				leaderElector, err := k8s.CreateLeaderElector(
+					leaderElectionLockName,
+					namespace,
+					leaderElectionId,
+					k8sClient,
+					leaderElectionConfiguration,
+					func(ctx context.Context) {
+						ctrl.Run(ctx, processorsCount)
+					})
+				if err != nil {
+					return err
+				}
+
+				go leaderElector.Run(context.Background())
+				<-context.Background().Done()
+			} else {
+				go ctrl.Run(context.Background(), processorsCount)
+				<-context.Background().Done()
 			}
 
-			go ctrl.Run(context.Background(), processorsCount)
-			<-context.Background().Done()
 			return nil
 		},
 	}
 	clientConfig = k8s.AddK8SFlagsToCmd(&command)
+	options.BindLeaderElectionFlags(leaderElectionConfiguration, command.Flags())
+	command.Flags().StringVar(&leaderElectionId, "leader-election-id", uuid.New().String(), "Instance id for leader election.")
```

Comment:
I think it is suitable we use pod name instead of uuid

Reply:
It's an option. But i wasn't found clear solution how to get current pod. Also documentation for k8s leader election contain examples with uuid. I will look on it one more time